AmplifyPlugins/Core/AWSPluginsCore/Model/Support/AuthRule+Extension.swift

```diff
@@ -7,6 +7,43 @@
 
 import Amplify
 
+protocol AWSAuthRuleProvider {
+    static var iam: AuthRuleProvider { get }
+    static var userPools: AuthRuleProvider { get }
+}
+
+extension AuthRuleProvider: AWSAuthRuleProvider {
+    private static func valueForIAM() -> String { "iam" }
+    private static func valueForCognitoUserPool() -> String { "userPools" }
+
+    public static var iam: AuthRuleProvider {
+        .custom(name: valueForIAM())
+    }
+
+    public static var userPools: AuthRuleProvider {
+        .custom(name: valueForCognitoUserPool())
+    }
+
+    public func toAWSAuthorizationType() throws -> AWSAuthorizationType {
+        var authType: AWSAuthorizationType
+        switch self {
+        case .apiKey:
+            authType = .apiKey
+        case .oidc:
+            authType = .openIDConnect
+        case .custom(name: let name) where name == AuthRuleProvider.valueForIAM():
+            authType = .awsIAM
+        case .custom(name: let name) where name == AuthRuleProvider.valueForCognitoUserPool():
+            authType = .amazonCognitoUserPools
+        case .custom(name: let name):
+            throw DataStoreError.unknown(
+                "Invalid AWS authorization provider in schema @auth rule: \(name)",
+                "Verify your schema.")
```

Comment:
thinking out loud... given the need to handle unknown cases and the `import AWSPluginsCore`, I wonder if we should drop the AWS abstraction and just define all the supported cases in the `AuthRuleProvider` enum itself.

Reply:
yeah, I agree it's probably easier to drop the AWS abstraction and simplify things a bit